website/pages/en/cookbook/grafting-hotfix.mdx

```diff
@@ -0,0 +1,184 @@
+         kind: ethereum/events
+         apiVersion: 0.0.7
+         language: wasm/assemblyscript
+         entities:
+           - Withdrawal
+         abis:
+           - name: Lock
+             file: ./abis/Lock.json
+         eventHandlers:
+           - event: Withdrawal(uint256,uint256)
+             handler: handleWithdrawal
+         file: ./src/lock.ts
+   features:
+     - grafting
+   graft:
+     base: QmBaseDeploymentID # Deployment ID of the failed subgraph
+     block: 6000000 # Last successfully indexed block
+   ```
+
+**Explanation:**
+
+- **Data Source Update**: The new subgraph points to 0xNewContractAddress, which may be a fixed version of the smart contract.
+- **Start Block**: Set to one block after the last successfully indexed block to avoid reprocessing the error.
+- **Grafting Configuration**:
+  - **base**: Deployment ID of the failed subgraph.
+  - **block**: Block number where grafting should begin.
+
+3. **Deployment Steps**
+
+   - **Update the Code**: Implement the hotfix in your mapping scripts (e.g., handleWithdrawal).
+   - **Adjust the Manifest**: As shown above, update the subgraph.yaml with grafting configurations.
+   - **Deploy the Subgraph**:
+     - Authenticate with the Graph CLI.
+     - Deploy the new subgraph using `graph deploy`.
+
+4. **Post-Deployment**
+   - **Verify Indexing**: Check that the subgraph is indexing correctly from the graft point.
+   - **Monitor Data**: Ensure that new data is being captured and the hotfix is effective.
+   - **Plan for Republish**: Schedule the deployment of a non-grafted version for long-term stability.
+
+## Key Considerations
+
+### When Not to Use Grafting
+
+While grafting is a powerful tool for deploying hotfixes quickly, there are specific scenarios where it should be avoided to maintain data integrity and ensure optimal performance.
+
+- **Incompatible Schema Changes**: If your hotfix requires altering the type of existing fields or removing fields from your schema, grafting is not suitable. Grafting expects the new subgraph’s schema to be compatible with the base subgraph’s schema. Incompatible changes can lead to data inconsistencies and errors because the existing data won’t align with the new schema.
+- **Significant Mapping Logic Overhauls**: When the hotfix involves substantial modifications to your mapping logic—such as changing how events are processed or altering handler functions—grafting may not function correctly. The new logic might not be compatible with the data processed under the old logic, leading to incorrect data or failed indexing.
+- **Deployments to The Graph Network**: Grafting is not recommended for subgraphs intended for The Graph’s decentralized network (mainnet). It can complicate indexing and may not be fully supported by all indexers, potentially causing unexpected behavior or increased costs. For mainnet deployments, it’s safer to re-index the subgraph from scratch to ensure full compatibility and reliability.
```

Comment:
```suggestion
- **Deployments to The Graph Network**: Grafting is not recommended for subgraphs intended for The Graph’s decentralized network (mainnet). It can complicate indexing and may not be fully supported by all Indexers, potentially causing unexpected behavior or increased costs. For mainnet deployments, it’s safer to re-index the subgraph from scratch to ensure full compatibility and reliability.
```